src/live_restore/live_restore_fs.c

```diff
@@ -1178,68 +1182,104 @@ __live_restore_fs_open_in_destination(WT_LIVE_RESTORE_FS *lr_fs, WT_SESSION_IMPL
 }
 
 /*
- * __live_restore_fs_open_file --
- *     Open a live restore file handle. This will: - If the file exists in the source, open it in
- *     both. - If it doesn't exist it'll only open it in the destination.
+ * __live_restore_setup_lr_fh_directory --
+ *     Populate a live restore file handle for a directory. Directories have special handling. If
+ *     they don't exist in the destination they'll be created immediately (but not their contents)
+ *     and immediately marked as complete. WiredTiger will never create or destroy a directory so we
+ *     don't need to think about tombstones.
  */
 static int
-__live_restore_fs_open_file(WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const char *name,
-  WT_FS_OPEN_FILE_TYPE file_type, uint32_t flags, WT_FILE_HANDLE **file_handlep)
+__live_restore_setup_lr_fh_directory(WT_SESSION_IMPL *session, WT_LIVE_RESTORE_FS *lr_fs,
```

Comment:
This diff is rather unhelpful. 
In the old code `__live_restore_fs_open_file` would setup the file handle only for files. I've left the common setup code in `__live_restore_fs_open_file` but pulled the file specific logic into `__live_restore_setup_lr_fh_directory`. I've also added `__live_restore_setup_lr_fh_directory` for the directory code and `__live_restore_fs_open_file` calls one of these subfunctions based on the `file_type` flag.